### PR TITLE
HS-842: Convert notification component endpoint from REST to gRPC

### PR DESCRIPTION
**Pull request by james8hut**
## Description
Convert notification component endpoint from REST to gRPC. This is to ensure consistency across the application, as to how services call each other.

## Jira link(s)
- https://issues.opennms.org/browse/HS-842

## Flagged for review
n/a

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

**Review by thomasbigger584 — APPROVED** [review]
general question: for consistency, should this service use JPA repository instead of the JdbcTemplate ?

**Comment by james8hut** [comment]
> general question: for consistency, should this service use JPA repository instead of the JdbcTemplate ?

Makes sense. Will create another story for that.

File: charts/opennms/templates/opennms/notification/notification-deployment.yaml [hunk 1]
```diff
@@ -70,3 +70,5 @@ spec:
           ports:
             - name: http
               containerPort: {{ .Values.OpenNMS.Notification.Port }}
```
Comment (cgorantla) [inline charts/opennms/templates/opennms/notification/notification-deployment.yaml:72]: Do we still need this ?

**Review by cgorantla — APPROVED** [review]
LGTM!

**Review by yangli2014 — APPROVED** [review]

File: charts/opennms/templates/opennms/notification/notification-deployment.yaml (hunk 1 above)
Comment (james8hut) [reply charts/opennms/templates/opennms/notification/notification-deployment.yaml]: No, will remove

**Review by thomasbigger584 — APPROVED** [review]

**Pull request merged**